pkg/virt-handler/network/network.go

```diff
@@ -0,0 +1,124 @@
+/*
+ * This file is part of the kubevirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017 Red Hat, Inc.
+ *
+ */
+
+package network
+
+import (
+	"fmt"
+	"strings"
+	"errors"
+
+	"github.com/jeevatkm/go-model"
+
+	"kubevirt.io/kubevirt/pkg/api/v1"
+	"kubevirt.io/kubevirt/pkg/logging"
+	"kubevirt.io/kubevirt/pkg/virt-handler/network/cniproxy"
+	"kubevirt.io/kubevirt/pkg/virt-handler/virtwrap"
+)
+
+const HostNetNS = "/proc/1/ns/net"
+
+type VirtualInterface interface {
+	Plug(domainManager virtwrap.DomainManager) error
+	Unplug(device string, domainManager virtwrap.DomainManager) error
+	GetConfig() (*v1.Interface, error)
+	DecorateInterfaceStatus() *v1.VMNetworkInterface
+}
+
+func GetInterfaceType(objName string) (VirtualInterface, error) {
+	switch objName {
+	case "PodNetworkInterface":
+		return new(PodNetworkInterface), nil
+	default:
+		return nil, errors.New("Invalid Interface type.")
+		}
+}
+
+func getInterfaceTypeFromVMStatus(vm *v1.VirtualMachine, device string) (VirtualInterface, error){
+	for _, iface := range vm.Status.Interfaces {
+		if iface.Device == device {
+			obj, err := GetInterfaceType(iface.Type)
+			if err != nil {
+				return nil, err
+			}
+			return obj, nil
+		}
+	}
+	return nil, fmt.Errorf("Failed to find device %s in known interfaces", device)
+}
+
+func UnPlugNetworkDevices(vm *v1.VirtualMachine, domainManager virtwrap.DomainManager) (error) {
```

Comment:
Some of the automated golang formatters we're using will try to auto format this line to remove the () around the (error) part of the function definition here.